src/coap_gnutls.c

```diff
@@ -68,6 +68,10 @@
                             GNUTLS_KEY_KEY_CERT_SIGN
 #endif /* GNUTLS_VERSION_NUMBER >= 0x030606 */
 
+#ifndef GNUTLS_CRT_RAW
```

Comment:
I would like to tie this to the proper GnuTLS version numbers as GNUTLS_CRT_RAWPK may or may not be defined either.

Reply:
@obgm There is no exact version number. According to commit 9838cc9a9b18ea5e3d94de1ba2f8bdd5013333d8 this define was removed since 3.6.4, and then restored after 3.6.12. 
So, using `ifndef` looks like simplest way to fix issue

Reply:
Version used in RPi default installation:
```
$ gnutls-cli -v
gnutls-cli 3.6.7
Copyright (C) 2000-2021 Free Software Foundation, and others, all rights reserved.
This is free software. It is licensed for use, modification and
redistribution under the terms of the GNU General Public License,
version 3 or later <http://gnu.org/licenses/gpl.html>


Please send bug reports to:  <bugs@gnutls.org>

```

Reply:
Ah, I see what they did. So, yes, you are correct that the `#ifndef` is the way go.